File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        st = strs[0]
        en = strs[-1]
        res = []
        for i in range(min(len(st),len(en))):
            if(st[i]!=en[i]):
                return ''.join(res)
            res.append(st[i])

        return ''.join(res)
        
                

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        string st = strs[0];
        string en = strs[strs.size()-1];
        string res ="";
        int mi = min(st.size(),en.size());
        for(int i =0;i<mi;i++){
            if(st[i]!=en[i]){
                return res;
            }
            res = res+st[i];
        }
        return res;
    }
};